Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/SamplerState.h

```diff
@@ -95,6 +95,9 @@ namespace AZ::RHI
         float m_mipLodMax = static_cast<float>(Limits::Image::MipCountMax);
         float m_mipLodBias = 0.0f;
         BorderColor m_borderColor = BorderColor::TransparentBlack;
+
+        bool operator==(const SamplerState& other) const { return GetHash() == other.GetHash(); }
+        bool operator!=(const SamplerState& other) const { return GetHash() != other.GetHash(); }
```

Comment:
These comparisons ignore hash collisions. It's very unlikely that those will occur, but they can still happpen